### PR TITLE
Added Query Hints for the Optimizer

### PR DESCRIPTION
**Pull request by pavelvelikhov**
### Changelog entry 

Query Hints for Cardinality and Join algorithms are added to the Cost Based Optimizer

### Changelog category 

* Improvement

### Additional information

Here is an example of how the new hints can be used in queries:

* Cardinality hints: a list of hints, each hint specifies the aliases or table names that are joined and specifies how the cardinality needs to be changed. Options for the 'op' field are: +,-,*,/ and #, where # means the value will be replaced by a constant
* Join hints: each hint specifies the aliases or table names that are joined and specifies the algorithm that needs to be picked for this specific join.

```
PRAGMA ydb.OptCardinalityHints = '[{"labels":["R"], "op":"#", "value":1}, {"labels":["R","S"], "op":"#", "value":10e6}]';
PRAGMA ydb.OptJoinAlgoHints = '[{"labels":["R","S"], "algo":"GraceJoin"}]';

SELECT *
FROM `/Root/R` as R
    INNER JOIN
        `/Root/S` as S
    ON R.id = S.id
```

### CODE REVIEW

File: ydb/library/yql/providers/dq/planner/execution_planner.cpp
Comment (pavelvelikhov) [inline ydb/library/yql/providers/dq/planner/execution_planner.cpp:1]: @zverevgeny Можешь посмотреть, нормальное изменение?

**Review by pashandor789 — DISMISSED** [review]

**Review by resetius — DISMISSED** [review]

**Review by pashandor789 — APPROVED** [review]

File: ydb/library/yql/providers/dq/planner/execution_planner.cpp [hunk 1]
```diff
@@ -40,6 +40,17 @@ using namespace NKikimr::NMiniKQL;
 
 using namespace Yql::DqsProto;
 
+namespace {
+    TString RemoveAliases(TString attributeName) {
+        for (size_t i = attributeName.size() - 1; i>0; i--) {
```
Comment (yumkam) [inline ydb/library/yql/providers/dq/planner/execution_planner.cpp:45]: nit: `.rfind('.')`?

Comment (pavelvelikhov) [reply ydb/library/yql/providers/dq/planner/execution_planner.cpp:45]: А-а, чего-то какие-то проблемы с find были, попробую отдельный PR с улучшением сделать
Сейчас хочется закомитить пока конфликты не приехали, большой PR получился

**Review by resetius — APPROVED** [review]

File: ydb/library/yql/providers/dq/planner/execution_planner.cpp (hunk 1 above)
Comment (resetius) [reply ydb/library/yql/providers/dq/planner/execution_planner.cpp:45]: size_t -> int

**Pull request merged**